FreeRTOS/Test/CBMC/proofs/Queue/QueueSemaphoreTake/QueueSemaphoreTake_harness.c

```diff
@@ -32,57 +32,26 @@
 #include "tasksStubs.h"
 #include "cbmc.h"
 
-BaseType_t state;
-QueueHandle_t xQueue;
-BaseType_t counter;
-
-BaseType_t xTaskGetSchedulerState(void)
-{
-	return state;
-}
-
-void vTaskInternalSetTimeOutState( TimeOut_t * const pxTimeOut )
-{
-	/* QueueSemaphoreTake might be blocked to wait for
-	   another process to release a token to the semaphore.
-	   This is currently not in the CBMC model. Anyhow,
-	   vTaskInternalSetTimeOutState is set a timeout for
-	   QueueSemaphoreTake operation. We use this to model a successful
-	   release during wait time. */
-	UBaseType_t bound;
-	__CPROVER_assume((bound >= 0 && xQueue->uxLength >= bound));
-	xQueue->uxMessagesWaiting = bound;
-}
-
 void harness()
 {
-	/* Init task stub to make sure that the third loop iteration
-	simulates a time out */
-	vInitTaskCheckForTimeOut(0, 3);
+	QueueHandle_t xQueue = xUnconstrainedMutex();
 
-	xQueue = xUnconstrainedMutex();
 	TickType_t xTicksToWait;
-
-	if(state == taskSCHEDULER_SUSPENDED){
-		xTicksToWait = 0;
-	}
-	if (xQueue) {
-		/* Bounding the loop in prvUnlockQueue to
-		   PRV_UNLOCK_QUEUE_BOUND. As the loop is not relevant
-		   in this proof the value might be set to any
-		   positive 8-bit integer value. We subtract one,
-		   because the bound must be one greater than the
-		   amount of loop iterations. */
-		__CPROVER_assert(PRV_UNLOCK_QUEUE_BOUND > 0, "Make sure, a valid macro value is chosen.");
-		xQueue->cTxLock = PRV_UNLOCK_QUEUE_BOUND - 1;
-		xQueue->cRxLock = PRV_UNLOCK_QUEUE_BOUND - 1;
-		((&(xQueue->xTasksWaitingToReceive))->xListEnd).pxNext->xItemValue = nondet_ticktype();
-
-		/* This assumptions is required to prevent an overflow in l. 2057 of queue.c
-		   in the prvGetDisinheritPriorityAfterTimeout() function. */
-		__CPROVER_assume( (
-		  ( UBaseType_t ) listGET_ITEM_VALUE_OF_HEAD_ENTRY( &( xQueue->xTasksWaitingToReceive ) )
-		   <= ( ( UBaseType_t ) configMAX_PRIORITIES)));
-		xQueueSemaphoreTake(xQueue, xTicksToWait);
-	}
+	__CPROVER_assume( xState != taskSCHEDULER_SUSPENDED || xTicksToWait == 0 );
+
+	/* This is for loop unwinding. */
+	__CPROVER_assume( xQueue->cTxLock = LOCK_BOUND - 1 );
+	__CPROVER_assume( xQueue->cRxLock = LOCK_BOUND - 1 );
+
+	/* Volatile member. */
+	((&(xQueue->xTasksWaitingToReceive))->xListEnd).pxNext->xItemValue = nondet_ticktype();
```

Comment:
Note for Mark -- this is a todo for me. Will come back to this in the next round.